File: backend/cotacoes-service/src/main/java/com/quotare/cotacoes/domain/Indicador.java
package com.quotare.cotacoes.domain;

import io.quarkus.hibernate.orm.panache.PanacheEntityBase;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

import java.time.Instant;

@Entity
@Table(name = "indicador")
public class Indicador extends PanacheEntityBase {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long id;

    public String codigo;

    public String nome;

    @Enumerated(EnumType.STRING)
    public FonteDados fonte;

    public Boolean ativo = true;

    @Column(name = "criado_em", insertable = false, updatable = false)
    public Instant criadoEm;

    @Column(name = "atualizado_em", insertable = false, updatable = false)
    public Instant atualizadoEm;
}

File: frontend/src/types/indicador.ts
import type { FonteDados } from './comum'

export interface Indicador {
  id: number
  codigo: string
  nome: string
  fonte: FonteDados
  ativo: boolean
  criadoEm: string
  atualizadoEm: string
}

export interface CriarIndicadorRequest {
  codigo: string
  nome: string
  fonte: FonteDados
}

export interface AtualizarIndicadorRequest {
  codigo: string
  nome: string
  fonte: FonteDados
  ativo: boolean
}

export interface IndicadorResumo {
  id: number
  codigo: string
  nome: string
}
